### ISSUE TITLE
Install doesn't work under Intune Autopilot

### ISSUE THREAD

**Issue by artem-shaturskyi** [Squirrel/Squirrel.Windows#1825]
**Description**
Squirrel based per user installers don't work under Intune Autopilot

**Steps to recreate**
1. Create Intune application
2. Installation command line <installer> -s
3. Configure Autopilot for the device
4. Enroll the device

**Expected behavior**
Application successfully installed

**Actual behavior**
The installation seems to even cannot start. Log file not created.
The installer returns error code -1073741819
In the Event Viewer Exception code: 0xc0000005

**Comment by timon-michel-scopevisio** [comment]
Just had the same issue, have you found any more information? Have you found a Workaround?
Intune reports "Access denied" while i can see the same Event viewer Exception code.
I can, however report that it only sometimes doesn't work. I have deployed 2 devices now on autopilot and 1 installed fine while another has this issue.
I didn't had any issues until i started migrating to AzureAD Hybrid. For me that feels like some kind of race condition.

**Comment by artem-shaturskyi** [comment]
No, unfortunately, I didn't find a workaround. After some time I gave up and now it is a "limitation" in our deployment environment.

**Comment by anaisbetts** [comment]
I don't know how InTune works but I suspect it is trying to execute the installer as a service account, you probably need to talk to your application vendor to get an MSI instead. Alternatively, every Squirrel app is fundamentally a portable app, so you could just copy `%LocalAppData%\YourApp` to the user's `%LocalAppData%` then run `%LocalAppData%\YourApp\YourApp.exe --squirrel-install 1.2.3` (where 1.2.3 is the version number). 

If you have more documentation about how InTune works I could provide a better recommendation, I don't really know what it is doing internally

**Comment by artem-shaturskyi** [comment]
The applications run with current user permissions - there is such possibility in Intune. Unfortunately, not all vendors provides alternative installers. Yes, it is possible to extract files and trigger executable in the script, etc., but question why Squirrel doesn't run in usual way, and how to make it run.

**Comment by anaisbetts** [comment]
I have no idea how to make Intune work with Squirrel, sorry! I cannot speculate as to why Intune was coded to only work with "the usual way" installers, because I did not write Intune.
